src/models/matches.rs

```diff
@@ -49,20 +63,44 @@ impl Match {
   ) -> Self {
     Self {
       matched_string,
-      range: Range {
-        start_byte: range.start_byte,
-        end_byte: range.end_byte,
-        start_point: Point {
-          row: range.start_point.row,
-          column: range.start_point.column,
-        },
-        end_point: Point {
-          row: range.end_point.row,
-          column: range.end_point.column,
-        },
-      },
+      range: Range::from(range),
       matches,
+      associated_comma: None,
+      associated_comments: Vec::new(),
+    }
+  }
+  ///
+  /// Returns the first and last associated ranges for the match.
+  /// If there are no associated ranges, returns the range of the match itself.
+  fn get_first_and_last_associated_ranges(&self) -> (Range, Range) {
+    // Sort all the associated ranges
+    let associated_ranges = [
+      self.associated_comma().iter().collect_vec(),
+      self.associated_comments().iter().collect_vec(),
+    ]
+    .concat()
+    .iter()
+    .sorted()
+    .cloned()
+    .collect_vec();
+    let start_range = associated_ranges.first().cloned().unwrap_or(&self.range);
+    let end_range = associated_ranges.last().cloned().unwrap_or(&self.range);
+    (*start_range, *end_range)
+  }
+
+  /// Merge the associated matches of the given match into the current match.
+  /// It basically extends the range to include the first and last associated match of the given match.
+  pub(crate) fn expand_to_associated_matches(&mut self, code: &String) {
+    let (start_range, end_range) = self.get_first_and_last_associated_ranges();
+    if start_range.start_byte < self.range.start_byte {
```

Comment:
Is there a case where `associated_ranges.first()` in `get_first_and_last_associated_ranges()` exists and yet it is after `self.range.start_byte`? Rather than before or equal? (Trying to figure out if the conditional checks here are needed or not.